### PR TITLE
Adding Microsoft.Extensions.Telemetry README

### PR DESCRIPTION
**Pull request by joperezr**
Adding README for Microsoft.Extensions.Telemetry and making one small clarification on one of the Compliance READMEs to explain how RedactorProviders should be used.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4671)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Compliance.Redaction/README.md [hunk 1]
```diff
@@ -31,18 +31,26 @@ public static IServiceCollection AddRedaction(this IServiceCollection services,
...
+    // Assigns a fallback redactor to use when processing classified data for which no specific redactor has been registered. 
+    // The `ErasingRedactor` is the default fallback redactor. If no redactor is configured for a data classification then the data will be erased.
+    redactionBuilder.SetFallbackRedactor<MyFallbackRedactor>();
+});
+
+// Using a custom redactor provider:
+builder.Services.AddRedaction(redactionBuilder =>
+{
+    redactionBuilder.Services.AddSingleton<IRedactorProvider, MyRedactorProvider>();
+});
```
Comment (Tratcher) [inline src/Libraries/Microsoft.Extensions.Compliance.Redaction/README.md:51]:
```suggestion
// Using a custom redactor provider:
builder.Services.AddSingleton<IRedactorProvider, MyRedactorProvider>();
builder.Services.AddRedaction(redactionBuilder => { });
```

**Review by Tratcher — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Telemetry/README.md [hunk 2]
```diff
@@ -18,6 +18,65 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage
+
+### Service Log Enrichment
+
+Enriches logs with application-specific information based on `ApplicationMetadata information. The bellow calls will add the service log enricher to the service collection.
```
Comment (sebastienros) [inline src/Libraries/Microsoft.Extensions.Telemetry/README.md:25]:
```suggestion
Enriches logs with application-specific information based on `ApplicationMetadata` information. The bellow calls will add the service log enricher to the service collection.
```

File: src/Libraries/Microsoft.Extensions.Telemetry/README.md [hunk 3]
```diff
@@ -18,6 +18,65 @@ Or directly in the C# project file:
...
+    options.ApplicationName = true;
+    options.BuildVersion = true;
+    options.DeploymentRing = true;
+    options.EnvironmentName = true;
+});
+```
+
+### Latency Monitoring
+
+Provides tools for latency data collection and export, particularly to console outputs.
```
[the remark this answers is not in the document]
Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.Telemetry/README.md]: Done. Also added an example.

File: src/Libraries/Microsoft.Extensions.Telemetry/README.md [hunk 4]
```diff
@@ -18,6 +18,90 @@ Or directly in the C# project file:
...
+
+builder.Services.AddServiceLogEnricher(); // <- This call is required in order for the enricher to be added into the service collection.
+
+// Enable log redaction
+builder.EnableRedaction(options =>
+{
+    options.ApplyDiscriminator = true;
+});
+
+builder.Services.AddRedaction(); // <- This call is required in order for the redactor provider to be added into the service collection.
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Telemetry/README.md:102]: IIRC this call requires `Microsoft.Extensions.Compliance.Redaction` to be referenced

File: src/Libraries/Microsoft.Extensions.Telemetry/README.md [hunk 5]
```diff
@@ -18,6 +18,90 @@ Or directly in the C# project file:
...
+app.UseRequestLatencyTelemetry();
+```
+
+### Logging Enhancements
+
+Offers additional logging capabilities like stack trace capturing, exception message inclusion, and log redaction.
+
+```csharp
+// Enable log enrichment.
+builder.EnableEnrichment(options =>
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Telemetry/README.md:86]: The extension method should be applied on an `ILoggingBuilder`

Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Telemetry/README.md:86]: Same applies to `EnableRedaction()` call

**Pull request merged**